src/components/member/tabs/quote-tab/quote-activation.tsx

```diff
@@ -1,31 +1,24 @@
-import { useMutation } from '@apollo/react-hooks'
-import { Quote } from 'api/generated/graphql'
-import { gql } from 'apollo-boost'
+import { Contract, Quote } from 'api/generated/graphql'
 import { BaseDatePicker } from 'components/shared/inputs/DatePicker'
-import { QUOTES_QUERY } from 'graphql/use-quotes'
 import { Button } from 'hedvig-ui/button'
 import * as React from 'react'
 import { Checkbox } from 'semantic-ui-react'
 import { noopFunction } from 'utils'
+import {
+  addAgreementFromQuoteOptions,
+  useAddAgreementFromQuote,
+} from '../../../../graphql/use-add-agreement-from-quote'
+import { useContracts } from '../../../../graphql/use-contracts'
 import { BottomSpacerWrapper, ErrorMessage } from './common'
 
-const ACTIVATE_MUTATION = gql`
-  mutation ActivateQuote(
-    $id: ID!
-    $activationDate: LocalDate!
-    $terminationDate: LocalDate
-  ) {
-    activateQuote(
-      id: $id
-      activationDate: $activationDate
-      terminationDate: $terminationDate
-    ) {
-      id
-      originatingProductId
-      signedProductId
-    }
-  }
-`
+const getContract = (contracts, quote): Contract => {
+  return contracts.find((contract) =>
+    contract.agreements.flatMap(
+      (agreement) => agreement.id === quote.originatingProductId,
+    ),
```

Comment:
If this works that's awesome, personally I'd do something like this
```suggestion
    contract.agreements.some(
      (agreement) => agreement.id === quote.originatingProductId,
    ),
```